addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClientV2.java

```diff
@@ -95,113 +122,105 @@ private LocationsStatus requestLocationsStatus() {
 
     @Override
     public boolean login() {
-        boolean success = false;
-        try {
-            // Building the HTTP request discretely here, as it is the only one with a deviant content type
-            Request request = httpClient.newRequest(EvohomeApiConstants.URL_V2_AUTH);
-            request.method(HttpMethod.POST);
-            request.header("Authorization", "Basic YjAxM2FhMjYtOTcyNC00ZGJkLTg4OTctMDQ4YjlhYWRhMjQ5OnRlc3Q=");
-            request.header("Accept", "application/json, application/xml, text/json, text/x-json, text/javascript, text/xml");
-
-            String data = "Username=" + URLEncoder.encode(configuration.username, "UTF-8") + "&"
-                    + "Password=" + URLEncoder.encode(configuration.password, "UTF-8") + "&"
-                    + "Host=rs.alarmnet.com%2F&"
-                    + "Pragma=no-cache&"
-                    + "Cache-Control=no-store+no-cache&"
-                    + "scope=EMEA-V1-Basic+EMEA-V1-Anonymous+EMEA-V1-Get-Current-User-Account&"
-                    + "grant_type=password&"
-                    + "Content-Type=application%2Fx-www-form-urlencoded%3B+charset%3Dutf-8&"
-                    + "Connection=Keep-Alive";
-
-            request.content(new StringContentProvider(data), "application/x-www-form-urlencoded");
-
-            ContentResponse response = request.send();
-            if (response.getStatus() == 200) {
-                String reply = response.getContentAsString();
-                success = true;
-                apiAccess.setAuthentication(new Gson().fromJson(reply, Authentication.class));
-            }
-        } catch (Exception e) {
-            apiAccess.setAuthentication(null);
-            logger.error("Authorization failed", e);
-        }
+       boolean success = authenticate();
 
         // If the authentication succeeded, gather the basic intel as well
         if (success == true) {
-            useraccount = requestUserAccount();
-            locations   = requestLocations();
+            useraccount        = requestUserAccount();
+            locations          = requestLocations();
```

Comment:
It's called here:
https://github.com/Nebula83/openhab2-addons/blob/httpClient_refactor/addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeGatewayHandler.java#L134

I could change the behavior so that the gateway only changes to ONLINE after the first successful update. That still doesn't solve that the discovery service is registered before the gateway is online, but at least it gives a visual indication that you shouldn't just yet press discover.

We could make that nicer: first deliver back the gateway, then nothing until the gateway is online then register the discovery service, but I wouldn't make that a priority for now.